neurodamus/report.py

```diff
@@ -158,12 +158,12 @@ def add_synapse_report(self, cell_obj, point, vgid, pop_name="default", pop_offs
         if not self.use_coreneuron:
             # Prepare the report for the cell
             self.report.AddNode(gid, pop_name, pop_offset)
-            for synapse in synapse_list:
-                try:
+            try:
+                for synapse in synapse_list:
                     var_ref = getattr(synapse, "_ref_" + variable)
                     self.report.AddVar(var_ref, synapse.synapseID, gid, pop_name)
-                except AttributeError:
-                    raise AttributeError(f"Variable '{variable}' not found at '{synapse.hname()}'.")
+            except AttributeError:
+                raise AttributeError(f"Variable '{variable}' not found at '{synapse.hname()}'.")
```

Comment:
why did you invert these 2? Since it raises an exception I think the final effect is the same so I suspect it is just for performance but I prefer to ask

Reply:
yeah, it's a perf thing that linting noticed, which is pretty cool, imo.